Content.Shared/PLC/Systems/PLCInterpreter.cs

```diff
@@ -0,0 +1,438 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+
+namespace Spessembly
+{
+    public class PLCInterpreter
+    {
+        private const char pinPrefix = 'p';
+        private const char registerPrefix = 'r';
+        private const char immediatePrefix = '#';
+        private const char addressStart = '[';
+        private const char addressEnd = ']';
+
+        public const int HIGH = 100;
+        public const int LOW = 0;
+
+        private string[] code = Array.Empty<string>();
```

Comment:
ideally these should be non-string as string comparisons take a LOT of computing power, something like:
```cs
private string code = "";
private string instructions = Array.Empty<PLCInstruction>();

// ...

public class PLCInstruction
{
    // yml hooks??? idfk maybe i'm getting WAY too inspired
    public string Name = "Invalid";
    public string Opcode = "nop";
    
    public virtual void Run(PLCVar[] args)
    {
        //Logger.Info("Ran instruction " + self.Name);
        return;
    }
}

public class AddI : PLCInstruction
{
    Name = "Add";
    Opcode = "add";
    
    public override void Run(PLCVar[] args)
    {
        if (args.Length < 3)
            return;
        args[2].SetValue(args[0].Value() + args[1].Value())
        return;
    }
}
```

Reply:
the code[] array is the actual text typed by the player, so I'd need to load it all in and then convert it to a separate array with instructions. Still have to check the strings, but then I do it only once instead of every step.
Probably a good idea, but I'm going to focus on getting it to work, and then optimising it after a working version is achieved

Reply:
optionally throw NMIs instead of returning void lmao